Edit and add to List Filter partial string matching (#2163)
temporalio/documentation: docs-src/concepts/what-is-a-list-filter.md

### Partial string match

> #### Text

Reviewer: Usually we try not to put a header right after another header.
Perhaps some contextualizing copy between them?

#### Text

<!-- note: advanced vis features will be supported in SQL upon the release of v1.20.-->
Search Attributes of type `Text` are broken up into words that match with the `=` operator.

For example, if you have a custom Search Attribute named `Description` of `Text` type with the value of "The quick brown fox jumps over the lazy dog", searching for `Description='quick'` or `Description='fox'` will successfully return the Workflow.
However, partial word searches such as `Description='qui'` or `Description='laz'` will not return the Workflow.

> This is because [Elasticsearch's tokenizer](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-standard-tokenizer.html) is configured to return complete words as tokens.

Reviewer: do we want to completely drop this example as well?
Having 2 might be good - and a link to the tokenizer is OK I think?

Author: Adding the tokenizer link. I think this example is the same as my example, I just used `my-business-id-foobar` and `my business id foobar` over `The quick brown fox jumps over the lazy dog`, since it seemed more common and demonstrates tokenization more

> For example, if you have a custom `Text` Search Attribute named `Description` with either of the following values:

Reviewer 2: ```suggestion
For example, if you have a custom `Text` Search Attribute named `Description` with either of the following values—
```
This and the related following edits are an informal approach to the grammar. Another approach is to make each intro line a complete sentence—entirely valid but does add words.

```

> then the following List Filter will match:

Reviewer 2: ```suggestion
—then the following List Filter matches—
```

```
Description = 'foobar'
```

> but a partial word will not:

Reviewer 2: ```suggestion
—but a partial word does not:
```

```
// Doesn't match
Description = 'foo'
```

#### Keyword

> For Search Attributes of type `Keyword` like `WorkflowId`, the only type of partial string matching that works is using BETWEEN for suffixes.
> For example:

Reviewer 2: ```suggestion
For Search Attributes of type `Keyword` like `WorkflowId`, the only kind of partial string matching that works is using BETWEEN for suffixes.
```

```
WorkflowId BETWEEN "order-" AND "order-~"
```

> would match WorkflowIds that have characters after `order-` with lower ASCII values than `~` (126, the highest-value printable character), such as the following:

Reviewer 2: ```suggestion
The preceding example matches WorkflowIds that have characters after `order-` with ASCII values lower than `~` (126, the highest-value printable character), such as the following:
```

Author: what about "This example matches"? "preceding" feels strange to me and makes me think it's referring to the one before the current example.

```
order-
order-1234
order-abracadabra
```

> It would not match `order-~~`.

Reviewer 2: ```suggestion
It does not match `order-~~`.
```
